### ISSUE TITLE
Remove commas for locations without States

### ISSUE THREAD

**Issue by RyanTG** [bpoore/pbm-react#405]
I'll add more to this later - placeholder so I don't forget.

If I'm remembering correctly, I did a search and one of the results in the list of results was for a location that didn't include a state (which is a valid case, as not all countries have states), but it was formatted wrong. I think it was "(City_name, )". We need to remove the comma and space if there isn't a state. We should check for all uses of the state field. As for venue examples, many venues in Finland do not have a state, such as "Flipperikellari"

**Comment by bpoore** [comment]
https://github.com/bpoore/pbm-react/commit/3f55a8f54c40fef2510eabe6a8f184f906f24ffb

**Comment by bpoore** [comment]
The only other thing I saw when auditing where we are using a city/state combo was in [Events](https://github.com/bpoore/pbm-react/blob/master/app/screens/Events.js#L82). We are referencing the state in the current map extent to determine the address to try to geocode. This seems like an edge-case, but we could end up geocoding more than necessary when the map extent is in an area without country states. Do we want to worry about this?

**Comment by RyanTG** [comment]
![6E7F4F43-F414-407A-9F6C-E91F122D7AD2](https://user-images.githubusercontent.com/632682/143720904-5bd34895-f0ea-4abe-baff-f281f88e083e.png)

search results also have an issue.

**Comment by RyanTG** [comment]
![8DF49244-3BE6-4287-A9D7-2E9F4A51795C](https://user-images.githubusercontent.com/632682/143720973-89e65ef2-0bd7-470b-a132-4bad277e984e.jpeg)

and here there appears to be an extra empty space. Same for List:

![573E4788-8FBB-4FCE-9FBE-FDEB53835508](https://user-images.githubusercontent.com/632682/143720994-81924e9d-b8b7-4d18-a640-e05380fd84ad.jpeg)

**Comment by bpoore** [comment]
I did miss the location details screen! Good catch. Pushed up a fix for that.

I had just done an audit in the app where we are putting together `city, state`. The first issue found is coming directly from the API.

<img width="419" alt="Screen Shot 2021-11-27 at 12 42 27 PM" src="https://user-images.githubusercontent.com/7720282/143722670-bf1ee7ab-3810-4547-bd05-060e1b53da08.png">

[PBM Issue](https://github.com/scottwainstock/pbm/issues/1325) to address the the search issue with the API

After the original commit I pushed up, the location list doesn't quite look like what was posted in your screenshot. I'm seeing:
![Simulator Screen Shot - iPhone 13 Pro - 2021-11-27 at 12 44 41](https://user-images.githubusercontent.com/7720282/143722801-a962ff63-d420-495b-a4a6-bab5e51a1b45.png)
